src/vitabel/vitals.py

```diff
@@ -439,10 +439,10 @@ def _add_single_dict(
         self,
         source: dict[str, Any],
         name: str,
-        metadata: dict = {},
         time_start=None,
-        datatype: str = "channel",
+        datatype: Literal["channel", "label", "interval_label"] = "channel",
         anchored_channel: Channel | None = None,
+        metadata: dict = {},
```

Comment:
FYI, @sorlob (and not your fault since it was like that before already): the empty dictionary is a dangerous default value since it is only created once, and any changes made to the argument by our code will modify this initial value. (See the "important warning" at https://docs.python.org/3/tutorial/controlflow.html#default-argument-values.) This is the reason why I usually go for `None` as the default, with a check

```py
if fancy_arg is None:
    fancy_arg = {}
```

I've started a commit with some minor improvements to this PR earlier this morning and just wasn't able to finish it yet; I'll just push it directly to the main branch.

Reply:
Done in #104